resolver.go

```diff
@@ -373,11 +391,13 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 			}()
 			err = co.WriteMsg(query)
 			if err != nil {
+				r.concurrentQueryDec()
```

Comment:
But it looks like the decrement must be done only after `co.WriteMsg()` or `co.ReadMsg()` has been executed ? Then my comment does not apply.